addons/website/models/website.py

```diff
@@ -1341,3 +1341,83 @@ def _get_cached_values(self):
 
     def _get_cached(self, field):
         return self._get_cached_values()[field]
+
+    def _get_html_fields(self):
+        html_fields = [('ir_ui_view', 'arch_db')]
+        cr = self.env.cr
+        cr.execute(r"""
+            SELECT f.model, f.name
+                FROM ir_model_fields f
+                JOIN ir_model m ON m.id = f.model_id
+            WHERE f.ttype = 'html'
+                AND f.store = true
+                AND m.transient = false
+                AND f.model NOT LIKE 'ir.actions%'
+                AND f.model != 'mail.message'
+        """)
+        for model, name in cr.fetchall():
+            table = self.env[model]._table
+            if tools.table_exists(cr, table) and tools.column_exists(cr, table, name):
+                html_fields.append((table, name))
+        return html_fields
+
+    def _get_snippets_assets(self):
+        """Returns every snippet asset from the database in the form of a list :
+        [snippet_module, snippet_id, asset_version, asset_type, asset_id]
+        """
+        self.env.cr.execute(r"""
+            SELECT regexp_matches[1] AS snippet_module,
+                   regexp_matches[2] AS snippet_id,
+                   regexp_matches[3] AS asset_version,
+                   CASE WHEN regexp_matches[4]='scss' THEN 'css'
+                        ELSE regexp_matches[4]
+                   END AS asset_type,
+                   id AS asset_id
+            FROM
+                (SELECT REGEXP_MATCHES(PATH, '(\w*)\/.*\/snippets\/(\w*)\/(\d{3})\.(js|scss)'),
+                        id
+                FROM ir_asset) AS regexp;
+        """)
+        return self.env.cr.fetchall()
+
+    def _is_snippet_used(self, snippet_module, snippet_id, asset_version, asset_type, html_fields):
+        snippet_occurences = []
+        # We check every snippet template definition
+        snippet_template = self.env.ref('%s.%s' % (snippet_module, snippet_id), raise_if_not_found=False)
+        if not snippet_template:
+            # If the snippet asset has been defined in a module
+            # that does not define the snippet template, it must be
+            # an override from a website snippet
+            snippet_template = self.env.ref('website.%s' % (snippet_id), raise_if_not_found=False)
```

Comment:
Can't you order by ID the assets, and exclude the assets for the same snippet?
Then everything will be solved IMO
Eg you have an asset 000.js for s_popup in website and one in website_mass_mailing -> remove the duplicate 000.js for s_popup -> we know it is an overide if we already have a 000.js for s_popup, and the first one (odered by id) will be the correct one

Reply:
Ordering by asset ID allows for retrieving the module originally defining the snippet template, for an asset that is an override. With that we can keep the overriden assets in the loop and check if they target snippets that are used or not :+1:

Reply:
Cool :+1: , we juste need to do an extra query when we neeed to active or inactive an asset, to do the same to its "inherited assets", so for example, in case you are going to inactive the website's s_popup's 000.js file, you need to also do that for all s_popup 000.JS files, in this case the one in `website_mass_mailing`

Reply:
Damn x) When @rdeodoo pinged me in this thread I replied this but it was put as pending :grimacing: Probably does not ably anymore but:

> Not merged yet, at worst this will need to be rebase :shrug: I will not analyse this code right now so I don't know when this method is called and what module it is given in what contexte, but we should not suppose a snippet is defined in any base module (any one could define a base one and be overridden by another). Maybe searching the view by ref is not a good plan, what about by key with `like f"_.{snippet_id}"` (would be a pity to have a conflict for snippet names as they start by `s_`)

Reply:
Indeed it does not **ably** anymore ;-)
We find the original asset with a tweak in the sql -> Distinct + order by id -> return the original
Then on active update, we find all the overide and process them the same way